src/Spark.Engine/Service/AsyncFhirService.cs

```diff
@@ -318,18 +343,20 @@ public async Task<FhirResponse> HandleInteractionAsync(Entry interaction)
                 case Bundle.HTTPVerb.POST:
                     return await CreateAsync(interaction).ConfigureAwait(false);
                 case Bundle.HTTPVerb.DELETE:
-                    var resourceStorage = GetFeature<IResourceStorageService>();
-                    var current = await resourceStorage.GetAsync(interaction.Key.WithoutVersion()).ConfigureAwait(false);
-                    if (current != null && current.IsPresent)
                     {
-                        return await DeleteAsync(interaction).ConfigureAwait(false);
+                        var resourceStorage = GetFeature<IResourceStorageService>();
+                        var current = await resourceStorage.GetAsync(interaction.Key.WithoutVersion())
+                            .ConfigureAwait(false);
+                        if (current != null && current.IsPresent)
+                        {
+                            return await DeleteAsync(interaction).ConfigureAwait(false);
+                        }
                     }
-                    // FIXME: there's no way to distinguish between "successfully deleted"
-                    // and "resource not deleted because it doesn't exist" responses, all return NoContent.
-                    // Same with Delete method above.
-                    return Respond.WithCode(HttpStatusCode.NoContent);
+                    return Respond.WithCode(HttpStatusCode.NotFound);
```

Comment:
@jjrdk This is the line that creates the issue for the integration tests. It needs to be set back to HttpStatusCode.NoContent for them to not fail.

Any reason you set it to HttpStatusCode.NotFound for this PR?

Reply:
Thanks so much @kennethmyhra for looking at this. I'll revert that change.
So much for boy scout cleanup 🙃